### DISCUSSION TITLE
Relationship Field not linked in Posts Table Pro

### DISCUSSION THREAD

**Discussion by slj1111** [pods-framework/pods#6691] [Ideas]
**Describe the bug**
I'm submitting this based on the discussion here: https://wordpress.org/support/topic/separating-pods-in-template/

I'm using a premium plugin called Posts Table Pro. It uses Pods to build the database. We're using it as a document library. Initially everything was working perfectly except the authors (which I added as a pod) were displaying alphabetically. The authors have to display in order of their contribution to the publication so it was suggested by Pods support that I enter the authors using a relationship field. I did that and it works perfectly on each separate publication page, but once I added the Custom Field to the Posts Table Pro shortcode it adds the author names but they are not linked.

**To Reproduce**
1. Go to http://test.otru.org/publications/

**Expected behavior**
 Ideally I would love to have the authors linked so that people can **sort the table based on their name**. This is what the date, publication type and keywords do now.

**Screenshots**

## Pods Version
2.7.9: 

## WordPress Environment

<details>
```
WordPress Version: 4.9.8

PHP Version: 5.5.36

MySQL Version: 5.5.50

Server Software: Apache

Your User Agent: Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.77 Safari/537.36

Session Save Path:

Session Save Path Exists: No

Session Save Path Writeable: No

Session Max Lifetime: 1440

Opcode Cache:

Apc: No
Memcached: No
OPcache: No
Redis: No
Object Cache:

APC: No
APCu: No
Memcache: No
Memcached: No
Redis: No
WPDB Prefix: wprw_

WP Multisite Mode: No

WP Memory Limit: 40M

Pods Network-Wide Activated: No

Pods Install Location: /home/otruorg/public_html/test.otru.org/wp-content/plugins/pods/

Pods Tableless Mode Activated: No

Pods Light Mode Activated: No

Currently Active Theme: Modernize Child Theme

Currently Active Plugins:

Google Fonts for WordPress: 1.3.1
Google Tag Manager for Wordpress: 1.9
Loginizer: 1.4.0
Pods - Custom Content Types and Fields: 2.7.9
Posts Table Pro: 2.1.1
Relevanssi: 4.1.0.1
WP Accessibility: 1.6.4
WP Super Cache: 1.6.4
```
</details>

## Pods Package Export (helpful!)

```
Copy and Paste the JSON Export from **Pods Admin, Migrate: Packages, Export** in WordPress admin here
```
{"meta":{"version":"2.7.9","build":1540823732},"pods":{"43":{"id":43,"name":"document_author","label":"Document Authors","description":"","type":"taxonomy","storage":"meta","object":"","alias":"","fields":[],"show_in_menu":"1","label_singular":"Document Author","public":"1","show_ui":"1","hierarchical":"1","rewrite":"1","rewrite_with_front":"1","rewrite_hierarchical":"1","capability_type":"default","capability_type_custom":"document_author","query_var":"0","sort":"0","built_in_post_types_custom_css":"0","built_in_post_types_customize_changeset":"0","built_in_post_types_documents":"1","built_in_post_types_gallery":"0","built_in_post_types_oembed_cache":"0","built_in_post_types_page":"0","built_in_post_types_personnel":"0","built_in_post_types_portfolio":"0","built_in_post_types_post":"0","built_in_post_types_price_table":"0","built_in_post_types_testimonial":"0","built_in_post_types_user_request":"0","built_in_post_types_attachment":"0","menu_location":"default","show_in_nav_menus":"1","show_tagcloud":"1","show_tagcloud_in_edit":"1","show_in_quick_edit":"1","show_admin_column":"0","pfat_enable":"0","pfat_run_outside_loop":"0","pfat_append_archive":"append","rest_enable":"0","read_all":"0","write_all":"0","built_in_post_types_authors":"0","built_in_post_types_staff":"0"},"18":{"id":18,"name":"document_category","label":"Document Categories","description":"","type":"taxonomy","storage":"meta","object":"","alias":"","fields":[],"show_in_menu":"1","label_singular":"Document Category","public":"1","show_ui":"1","hierarchical":"1","rewrite":"1","rewrite_with_front":"1","rewrite_hierarchical":"1","capability_type":"default","capability_type_custom":"document_category","query_var":"0","sort":"0","built_in_post_types_custom_css":"0","built_in_post_types_customize_changeset":"0","built_in_post_types_documents":"1","built_in_post_types_gallery":"0","built_in_post_types_oembed_cache":"0","built_in_post_types_page":"0","built_in_post_types_personnel":"0","built_in_post_types_portfolio":"0","built_in_post_types_post":"0","built_in_post_types_price_table":"0","built_in_post_types_testimonial":"0","built_in_post_types_user_request":"0","built_in_post_types_attachment":"0","menu_location":"default","show_in_nav_menus":"1","show_tagcloud":"1","show_tagcloud_in_edit":"1","show_in_quick_edit":"1","show_admin_column":"0","pfat_enable":"0","pfat_run_outside_loop":"0","pfat_append_archive":"append","rest_enable":"0","read_all":"0","write_all":"0","built_in_post_types_authors":"0","built_in_post_types_staff":"0"},"40":{"id":40,"name":"document_month","label":"Document Month","description":"","type":"taxonomy","storage":"meta","object":"","alias":"","fields":[],"show_in_menu":"1","label_singular":"Document Month","public":"1","show_ui":"1","hierarchical":"1","rewrite":"1","rewrite_with_front":"1","rewrite_hierarchical":"1","capability_type":"default","capability_type_custom":"document_month","query_var":"0","sort":"0","built_in_post_types_custom_css":"0","built_in_post_types_customize_changeset":"0","built_in_post_types_documents":"1","built_in_post_types_gallery":"0","built_in_post_types_oembed_cache":"0","built_in_post_types_page":"0","built_in_post_types_personnel":"0","built_in_post_types_portfolio":"0","built_in_post_types_post":"0","built_in_post_types_price_table":"0","built_in_post_types_testimonial":"0","built_in_post_types_user_request":"0","built_in_post_types_attachment":"0","menu_location":"default","show_in_nav_menus":"1","show_tagcloud":"1","show_tagcloud_in_edit":"1","show_in_quick_edit":"1","show_admin_column":"0","pfat_enable":"0","pfat_run_outside_loop":"0","pfat_append_archive":"append","rest_enable":"0","read_all":"0","write_all":"0","built_in_post_types_authors":"0","built_in_post_types_staff":"0"},"33":{"id":33,"name":"document_year","label":"Document Year","description":"","type":"taxonomy","storage":"meta","object":"","alias":"","fields":[],"show_in_menu":"1","label_singular":"Document Year","public":"1","show_ui":"1","hierarchical":"1","rewrite":"1","rewrite_with_front":"1","rewrite_hierarchical":"1","capability_type":"default","capability_type_custom":"document_year","query_var":"0","sort":"0","built_in_post_types_custom_css":"0","built_in_post_types_customize_changeset":"0","built_in_post_types_documents":"1","built_in_post_types_gallery":"0","built_in_post_types_oembed_cache":"0","built_in_post_types_page":"0","built_in_post_types_personnel":"0","built_in_post_types_portfolio":"0","built_in_post_types_post":"0","built_in_post_types_price_table":"0","built_in_post_types_testimonial":"0","built_in_post_types_user_request":"0","built_in_post_types_attachment":"0","menu_location":"default","show_in_nav_menus":"1","show_tagcloud":"1","show_tagcloud_in_edit":"1","show_in_quick_edit":"1","show_admin_column":"0","pfat_enable":"0","pfat_run_outside_loop":"0","pfat_append_archive":"append","rest_enable":"0","read_all":"0","write_all":"0","built_in_post_types_authors":"0","built_in_post_types_staff":"0"},"15":{"id":15,"name":"documents","label":"Documents","description":"","type":"post_type","storage":"meta","object":"","alias":"","fields":{"document_link":{"id":16,"name":"document_link","label":"document_link","description":"","help":"","class":"","type":"wysiwyg","weight":0,"pick_object":"","pick_val":"","sister_id":"","required":"0","wysiwyg_editor":"tinymce","wysiwyg_media_buttons":"1","wysiwyg_oembed":"0","wysiwyg_wptexturize":"1","wysiwyg_convert_chars":"1","wysiwyg_wpautop":"1","wysiwyg_allow_shortcode":"0","pick_post_status":["publish"],"admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"rest_read":"0","rest_write":"0","rest_pick_response":"array","rest_pick_depth":"2","unique":"0","wysiwyg_repeatable":"0","wysiwyg_allowed_html_tags":""},"publication_title":{"id":1052,"name":"publication_title","label":"Publication Title","description":"","help":"","class":"","type":"text","weight":1,"pick_object":"","pick_val":"","sister_id":"","required":"0","text_allow_shortcode":"0","text_allow_html":"0","text_allowed_html_tags":"strong em a ul ol li b i","text_max_length":"255","pick_post_status":["publish"],"admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"rest_read":"0","rest_write":"0","rest_pick_response":"array","rest_pick_depth":"2","unique":"0","text_repeatable":"0","text_placeholder":""},"related_authors":{"id":1057,"name":"related_authors","label":"Related Authors","description":"","help":"","class":"","type":"pick","weight":2,"pick_object":"post_type","pick_val":"authors","sister_id":"1059","required":"0","pick_format_type":"multi","pick_format_single":"dropdown","pick_format_multi":"list","pick_allow_add_new":"1","pick_taggable":"0","pick_show_icon":"1","pick_show_edit_link":"1","pick_show_view_link":"1","pick_limit":"0","pick_user_role":[],"pick_post_status":["publish"],"admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"rest_read":"0","rest_write":"0","rest_pick_response":"array","rest_pick_depth":"2","unique":"0","pick_select_text":"","pick_table_id":"","pick_table_index":"","pick_display":"","pick_where":"","pick_orderby":"","pick_groupby":""}},"show_in_menu":"1","label_singular":"Documents","public":"1","show_ui":"1","supports_title":"1","supports_editor":"1","publicly_queryable":"1","exclude_from_search":"0","capability_type":"post","capability_type_custom":"documents","capability_type_extra":"1","has_archive":"1","hierarchical":"0","rewrite":"1","rewrite_with_front":"1","rewrite_feeds":"0","rewrite_pages":"1","query_var":"1","can_export":"1","default_status":"draft","supports_author":"0","supports_thumbnail":"1","supports_excerpt":"1","supports_trackbacks":"0","supports_custom_fields":"1","supports_comments":"0","supports_revisions":"1","supports_page_attributes":"1","supports_post_formats":"1","built_in_taxonomies_category":"0","built_in_taxonomies_link_category":"0","built_in_taxonomies_personnel-category":"0","built_in_taxonomies_portfolio-category":"0","built_in_taxonomies_portfolio-tag":"0","built_in_taxonomies_post_tag":"0","built_in_taxonomies_price-table-category":"0","built_in_taxonomies_testimonial-category":"0","show_in_nav_menus":"1","show_in_admin_bar":"1","pfat_enable":"1","pfat_run_outside_loop":"0","pfat_append_single":"append","pfat_append_archive":"append","pfat_filter_archive":"the_content","rest_enable":"0","read_all":"0","write_all":"0","built_in_taxonomies_document_category":"1","built_in_taxonomies_document_tag":"1","built_in_taxonomies_document_year":"1","built_in_taxonomies_document_month":"1","built_in_taxonomies_document_author":"1","pfat_single":"PDF Template","pfat_filter_single":"the_content","has_archive_slug":"documents-archive"},"27":{"id":27,"name":"post","label":"Post","description":"","type":"post_type","storage":"meta","object":"post","alias":"","fields":{"document_link":{"id":28,"name":"document_link","label":"document_link","description":"","help":"","class":"","type":"text","weight":0,"pick_object":"","pick_val":"","sister_id":"","required":"0","text_allow_shortcode":"0","text_allow_html":"0","text_allowed_html_tags":"strong em a ul ol li b i","text_max_length":"255","pick_post_status":["publish"],"admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"rest_read":"0","rest_write":"0","rest_pick_response":"array","rest_pick_depth":"2","unique":"0","text_repeatable":"0","text_placeholder":""}},"show_in_menu":1,"pfat_enable":"1","pfat_run_outside_loop":"0","pfat_append_single":"append","pfat_filter_single":"the_content","pfat_append_archive":"append","pfat_filter_archive":"the_content","rest_enable":"0","read_all":"0","write_all":"0","pfat_single":"PDF Template","built_in_taxonomies_document_year":"0","built_in_taxonomies_document_category":"0","built_in_taxonomies_document_tag":"0","built_in_taxonomies_document_author2":"0","built_in_taxonomies_document_author3":"0","built_in_taxonomies_document_author4":"0"},"1046":{"id":1046,"name":"authors","label":"Publication Author(s)","description":"","type":"post_type","storage":"meta","object":"","alias":"","fields":{"publication_title":{"id":1059,"name":"publication_title","label":"Publication Title","description":"","help":"","class":"","type":"pick","weight":0,"pick_object":"post_type","pick_val":"documents","sister_id":"1057","required":"0","pick_format_type":"multi","pick_format_single":"dropdown","pick_format_multi":"list","pick_allow_add_new":"1","pick_taggable":"0","pick_show_icon":"1","pick_show_edit_link":"1","pick_show_view_link":"1","pick_limit":"0","pick_user_role":[],"pick_post_status":["publish"],"admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"rest_read":"0","rest_write":"0","rest_pick_response":"array","rest_pick_depth":"2","unique":"0","pick_select_text":"","pick_table_id":"","pick_table_index":"","pick_display":"","pick_where":"","pick_orderby":"","pick_groupby":""}},"show_in_menu":"1","label_singular":"Publication Author","public":"1","show_ui":"1","supports_title":"1","supports_editor":"1","publicly_queryable":"1","exclude_from_search":"0","capability_type":"post","capability_type_custom":"authors","capability_type_extra":"1","has_archive":"1","hierarchical":"0","rewrite":"1","rewrite_with_front":"1","rewrite_feeds":"0","rewrite_pages":"1","query_var":"1","can_export":"1","default_status":"draft","supports_author":"0","supports_thumbnail":"0","supports_excerpt":"1","supports_trackbacks":"0","supports_custom_fields":"1","supports_comments":"0","supports_revisions":"1","supports_page_attributes":"1","supports_post_formats":"1","built_in_taxonomies_category":"0","built_in_taxonomies_document_author":"0","built_in_taxonomies_document_category":"0","built_in_taxonomies_document_month":"0","built_in_taxonomies_document_tag":"0","built_in_taxonomies_document_year":"0","built_in_taxonomies_link_category":"0","built_in_taxonomies_personnel-category":"0","built_in_taxonomies_portfolio-category":"0","built_in_taxonomies_portfolio-tag":"0","built_in_taxonomies_post_tag":"0","built_in_taxonomies_price-table-category":"0","built_in_taxonomies_testimonial-category":"0","show_in_nav_menus":"1","show_in_admin_bar":"1","pfat_enable":"1","pfat_run_outside_loop":"0","pfat_single":"Authors","pfat_append_single":"append","pfat_filter_single":"the_content","pfat_append_archive":"append","pfat_filter_archive":"the_content","rest_enable":"0","read_all":"0","write_all":"0","pfat_archive":"Authors","has_archive_slug":"authors-archive"},"1186":{"id":1186,"name":"staff","label":"Staff","description":"","type":"post_type","storage":"meta","object":"","alias":"","fields":{"staff":{"id":1187,"name":"staff","label":"Staff","description":"","help":"","class":"","type":"wysiwyg","weight":0,"pick_object":"","pick_val":"","sister_id":"","required":"0","wysiwyg_editor":"tinymce","wysiwyg_media_buttons":"1","wysiwyg_oembed":"0","wysiwyg_wptexturize":"1","wysiwyg_convert_chars":"1","wysiwyg_wpautop":"1","wysiwyg_allow_shortcode":"0","pick_post_status":["publish"],"admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"rest_read":"0","rest_write":"0","rest_pick_response":"array","rest_pick_depth":"2","unique":"0","wysiwyg_repeatable":"0","wysiwyg_allowed_html_tags":""}},"show_in_menu":"1","label_singular":"Staff","public":"1","show_ui":"1","supports_title":"1","supports_editor":"1","publicly_queryable":"1","exclude_from_search":"0","capability_type":"post","capability_type_custom":"staff","capability_type_extra":"1","has_archive":"1","hierarchical":"0","rewrite":"1","rewrite_with_front":"1","rewrite_feeds":"0","rewrite_pages":"1","query_var":"1","can_export":"1","default_status":"draft","supports_author":"0","supports_thumbnail":"1","supports_excerpt":"1","supports_trackbacks":"0","supports_custom_fields":"1","supports_comments":"0","supports_revisions":"0","supports_page_attributes":"0","supports_post_formats":"0","built_in_taxonomies_category":"0","built_in_taxonomies_document_author":"0","built_in_taxonomies_document_category":"0","built_in_taxonomies_document_month":"0","built_in_taxonomies_document_tag":"0","built_in_taxonomies_document_year":"0","built_in_taxonomies_link_category":"0","built_in_taxonomies_personnel-category":"0","built_in_taxonomies_portfolio-category":"0","built_in_taxonomies_portfolio-tag":"0","built_in_taxonomies_post_tag":"0","built_in_taxonomies_price-table-category":"0","built_in_taxonomies_testimonial-category":"0","show_in_nav_menus":"1","show_in_admin_bar":"1","pfat_enable":"1","pfat_run_outside_loop":"0","pfat_single":"Staff","pfat_append_single":"append","pfat_filter_single":"the_content","pfat_append_archive":"append","pfat_filter_archive":"the_content","rest_enable":"0","read_all":"0","write_all":"0","pfat_archive":"Staff","has_archive_slug":"staff-archive"},"21":{"id":21,"name":"document_tag","label":"Subject","description":"","type":"taxonomy","storage":"meta","object":"","alias":"","fields":[],"show_in_menu":"1","label_singular":"Document Tag","public":"1","show_ui":"1","hierarchical":"1","rewrite":"1","rewrite_with_front":"1","rewrite_hierarchical":"1","capability_type":"default","capability_type_custom":"document_tag","query_var":"0","sort":"0","built_in_post_types_custom_css":"0","built_in_post_types_customize_changeset":"0","built_in_post_types_documents":"1","built_in_post_types_gallery":"0","built_in_post_types_oembed_cache":"0","built_in_post_types_page":"0","built_in_post_types_personnel":"0","built_in_post_types_portfolio":"0","built_in_post_types_post":"0","built_in_post_types_price_table":"0","built_in_post_types_testimonial":"0","built_in_post_types_user_request":"0","built_in_post_types_attachment":"0","menu_location":"default","show_in_nav_menus":"1","show_tagcloud":"1","show_tagcloud_in_edit":"1","show_in_quick_edit":"1","show_admin_column":"0","pfat_enable":"0","pfat_run_outside_loop":"0","pfat_append_archive":"append","rest_enable":"0","read_all":"0","write_all":"0","built_in_post_types_authors":"0","built_in_post_types_staff":"0"}},"templates":{"PDF Template":{"id":1026,"name":"PDF Template","code":"<H4>Author(s): <span>{@related_authors}<\/span><\/H4>\r\n<H4>Date: <span>{@document_month} {@document_year}<\/span><\/H4>\r\n<H4>Type of Publication: <span>{@document_category}<\/span><\/H4>\r\n\r\n[message_box title=\"Download this publication:\" color=\"blue\"]<img src=\"\/wp-content\/uploads\/2018\/10\/pdficon.png\" alt=\"\" width=\"31\" height=\"42\" class=\"alignleft size-full\" \/> <strong>{@document_link}<\/strong>[\/message_box]","type":"template","slug":"pdf-template","options":{"_edit_last":"1","_edit_lock":"1540398881:1","admin_only":"","restrict_capability":"","capability_allowed":"","pod_reference":"a:1:{s:3:\"pod\";s:9:\"documents\";}"}},"Authors":{"id":1077,"name":"Authors","code":"<div>Publications by this Author:<\/div>\r\n\r\n[if services_performed]\r\n  <ul>\r\n    [each services_performed]\r\n      <li><a href=\"{@document_link}\">{@related_authors.publication_title.publication_title}<\/a>\r\n    [each]\r\n  <\/ul>\r\n[else]\r\nThis staff member is not an author in any publication\r\n[\/if]\r\n      \r\n\r\nNOT WORKING","type":"template","slug":"authors","options":{"_edit_last":"1","_edit_lock":"1540412064:1","admin_only":"","restrict_capability":"","capability_allowed":"","pod_reference":"a:1:{s:3:\"pod\";s:9:\"documents\";}"}},"Staff":{"id":1188,"name":"Staff","code":"","type":"template","slug":"staff","options":{"_edit_last":"1","_edit_lock":"1540476096:1","pod_reference":"a:1:{s:3:\"pod\";s:5:\"staff\";}","admin_only":"","restrict_capability":"","capability_allowed":""}}}}

**Additional context**

**Possible Workaround**
If you have discovered a workaround, please include it below.

**Comment by jimtrue** [comment]
@slj1111 This would require some assistance from the Posts Table Pro plugin developers. Since that is a premium plugin, we don't have access to their code. If you can, as you have purchased the product, open a support ticket with Posts Table Pro and reference this GitHub issue. That way we can work with them to see if we can help them add support for our relationship fields into their product.

Thanks!

**Comment by slj1111** [comment]
Thank you Jim. I did add a ticket with them linking to this post but I don't think there is interest on their end. I suppose it will sit here until (and if) they want to look at it. Thank you again for all your help!

**Comment by jimtrue** [comment]
Did you get a response from them on your ticket? You can invite them to our Slack Chat as well (https://pods.io/chat/) and we can discuss with the developer. We might be able to add the functionality on our end, but it's usually better for them to have the code on their side.

**Comment by slj1111** [comment]
There was a response about not supporting that functionality. However, I think it would be good for them to incorporate, I can't be the only one finding that useful. Hopefully they will participate. Thanks for helping.
